### ISSUE TITLE
import catalog : server closed the connection unexpectedly

### ISSUE THREAD

**Issue by capitaine** [openlabs/magento_integration#125]
Installation went ok. No problem with website import. No problem with carriers import. But...

I imported "successfully" a catalog of about 100 products with no problems (excepted a categorization problem that will be reported separately)
But when I log out, then I get errors (see bellow after the first big error message). This happened on each of the 3 attempts to import catalog. The two first ones, I could neither login back to openerp and I had to reinstall everything. Now it seems I can log in and see the products, but I get the error on every log out. Mmmmm... Non, finally I can no more log in. I can if I don't close the browser, but if I close the browser and then reopen it, then I can not login. Instead, I get a 500 Internal Server Error and this message on the webpage:

{"message": "OpenERP Server Error", "code": 200, "data": {"debug": "Server Traceback (most recent call last):\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\session.py\", line 89, in send\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\netsvc.py\", line 292, in dispatch_rpc\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\service\web_services.py\", line 122, in dispatch\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\service\web_services.py\", line 359, in exp_list\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\sql_db.py\", line 484, in cursor\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\sql_db.py\", line 182, in **init**\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\sql_db.py\", line 377, in _locked\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\.\openerp\sql_db.py\", line 440, in borrow\n File \"psycopg2\__init__.pyc\", line 178, in connect\nOperationalError: server closed the connection unexpectedly\n\tThis probably means the server terminated abnormally\n\tbefore or while processing the request.\n\n\nClient Traceback (most recent call last):\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\http.py\", line 285, in dispatch\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py\", line 573, in index\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py\", line 98, in db_monodb_redirect\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py\", line 109, in db_redirect\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py\", line 90, in db_list\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\session.py\", line 30, in proxy_method\n File \"C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\session.py\", line 103, in send\nFault: <Fault server closed the connection unexpectedly\n\tThis probably means the server terminated abnormally\n\tbefore or while processing the request.\n: 'Traceback (most recent call last):\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\openerp\\\\addons\\\\web\\\\session.py\", line 89, in send\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\netsvc.py\", line 292, in dispatch_rpc\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\service\\\\web_services.py\", line 122, in dispatch\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\service\\\\web_services.py\", line 359, in exp_list\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\sql_db.py\", line 484, in cursor\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\sql_db.py\", line 182, in __init__\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\sql_db.py\", line 377, in _locked\\n File \"C:\\\\Program Files\\\\OpenERP 7.0-20130917-231058\\\\Server\\\\server\\\\.\\\\openerp\\\\sql_db.py\", line 440, in borrow\\n File \"psycopg2\\\\__init__.pyc\", line 178, in connect\\nOperationalError: server closed the connection unexpectedly\\n\\tThis probably means the server terminated abnormally\\n\\tbefore or while processing the request.\\n\\n'>\n", "fault_code": "server closed the connection unexpectedly\n\tThis probably means the server terminated abnormally\n\tbefore or while processing the request.\n", "type": "server_exception"}} 

Following errors are the one I get at the beginning of the problem (first logout after importing catalog)

web client side :

OpenERP Server Error

Client Traceback (most recent call last):
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\http.py", line 204, in dispatch
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py", line 764, in get_list
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py", line 129, in db_monodb
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py", line 109, in db_redirect
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py", line 90, in db_list
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\session.py", line 30, in proxy_method
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\session.py", line 103, in send

Server Traceback (most recent call last):
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server\openerp\addons\web\session.py", line 89, in send
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\netsvc.py", line 292, in dispatch_rpc
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line 122, in dispatch
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line 359, in exp_list
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 484, in cursor
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 182, in **init**
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 377, in _locked
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 440, in borrow
  File "psycopg2__init__.pyc", line 178, in connect
OperationalError: server closed the connection unexpectedly
    This probably means the server terminated abnormally
    before or while processing the request.

and this is  the openerp-server.log side

2013-09-20 22:57:57,717 1048 ERROR ? openerp.sql_db: Connection to the database failed
Traceback (most recent call last):
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 440, in borrow
  File "psycopg2__init__.pyc", line 178, in connect
OperationalError: server closed the connection unexpectedly
    This probably means the server terminated abnormally
    before or while processing the request.

2013-09-20 22:57:57,717 1048 ERROR ? openerp.netsvc: server closed the connection unexpectedly
    This probably means the server terminated abnormally
    before or while processing the request.
Traceback (most recent call last):
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\netsvc.py", line 292, in dispatch_rpc
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line 122, in dispatch
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line 359, in exp_list
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 484, in cursor
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 182, in **init**
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 377, in _locked
  File "C:\Program Files\OpenERP 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 440, in borrow
  File "psycopg2__init__.pyc", line 178, in connect
OperationalError: server closed the connection unexpectedly
    This probably means the server terminated abnormally
    before or while processing the request.

Is there, at least, a way to login to openerp without having to reinstall ?

**Comment by sharoonthomas** [comment]
Likely to be a problem with openerp running on windows. I doubt if the
system is sufficiently well tested on windows.

On Saturday, September 21, 2013, capitaine wrote:

> I imported a catalog of about 100 products with no problems (excepted a
> categorization problem reported separately)
> But when I log out, then I get following errors. This happened on each of
> the 3 attempts to import catalog. The two first ones, I could neither login
> back to openerp and I had to reinstall everything. Now it seems I can log
> in, but I get the error on every log out.
> 
> web client side :
> 
> OpenERP Server Error
> 
> Client Traceback (most recent call last):
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\http.py", line 204, in
> dispatch
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py",
> line 764, in get_list
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py",
> line 129, in db_monodb
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py",
> line 109, in db_redirect
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\controllers\main.py",
> line 90, in db_list
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\session.py", line 30,
> in proxy_method
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\session.py", line 103,
> in send
> 
> Server Traceback (most recent call last):
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server\openerp\addons\web\session.py", line 89,
> in send
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\netsvc.py", line 292, in
> dispatch_rpc
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line
> 122, in dispatch
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line
> 359, in exp_list
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 484, in cursor
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 182, in _init_
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 377, in *
> locked
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 440, in borrow
> File "psycopg2*_init__.pyc", line 178, in connect
> OperationalError: server closed the connection unexpectedly
> This probably means the server terminated abnormally
> before or while processing the request.
> 
> and openerp-server.log side
> 
> 2013-09-20 22:57:57,717 1048 ERROR ? openerp.sql_db: Connection to the
> database failed
> Traceback (most recent call last):
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 440, in borrow
> File "psycopg2__init__.pyc", line 178, in connect
> OperationalError: server closed the connection unexpectedly
> This probably means the server terminated abnormally
> before or while processing the request.
> 
> 2013-09-20 22:57:57,717 1048 ERROR ? openerp.netsvc: server closed the
> connection unexpectedly
> This probably means the server terminated abnormally
> before or while processing the request.
> Traceback (most recent call last):
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\netsvc.py", line 292, in
> dispatch_rpc
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line
> 122, in dispatch
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\service\web_services.py", line
> 359, in exp_list
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 484, in cursor
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 182, in _init_
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 377, in *
> locked
> File "C:\Program Files\OpenERP
> 7.0-20130917-231058\Server\server.\openerp\sql_db.py", line 440, in borrow
> File "psycopg2*_init__.pyc", line 178, in connect
> OperationalError: server closed the connection unexpectedly
> This probably means the server terminated abnormally
> before or while processing the request.
> 
> As this happened already with all 3 attempts for importing the catalog,
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/openlabs/magento_integration/issues/125
> .

## 

## 

Sharoon Thomas
Openlabs Technologies & Consulting (P) Limited

w: http://www.openlabs.co.in
m: +1 813.793.6736 (OPEN) Extn. 200
t: @sharoonthomas
- We Win When our Customers Win

**Comment by capitaine** [comment]
Perhaps, but the problem only occurs after import of magento catalog with magento_integration. The system worked fine with no problem before the import. I repeated the operation 3 times (with complete re-installation of openErp and postgresql) in order to verify. In last test, I logged out and logged in after each operation I was doing inside openERP before importing the catalog to see if the described problem could not be due to something else. I don't think it's a openERP/Windows issue. The +++ File "psycopg2_init__.pyc", line 178, in connect
OperationalError: server closed the connection unexpectedly +++ problem seems (after some googling) to be a python/postgres issue  but in any case it only appears after using the import catalog function of magento_integration. I can't test further as I can no more access the openERP login page : 500 internal server error :(
